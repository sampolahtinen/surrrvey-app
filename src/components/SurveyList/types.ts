import { AllSurveysResponse } from "../../api/types";

export interface SurveyListProps {
  surveys: AllSurveysResponse["surveys"];
}
