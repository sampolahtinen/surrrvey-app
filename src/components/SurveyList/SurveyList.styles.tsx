import styled from "styled-components";
import { colors } from "../../styles/colors";
import { animated } from "react-spring";
import { Link } from "react-router-dom";
import { breakPoints } from "../../styles/breakpoints";

export const SurveyListContainer = styled.div`
  width: 100%;
`;

export const List = styled.div``;

export const TagLine = styled.span`
  color: ${colors.grey};
  opacity: 0.3;
  @media (max-width: ${breakPoints.phone}) {
    display: block;
    margin-top: 4px;
  }
`;

const AnimatedLink = animated(Link);

export const Survey = styled(AnimatedLink)`
  display: flex;
  align-items: center;
  cursor: pointer;
  margin: 2.4rem 0;
  color: ${colors.darkGrey};
  &:hover {
    transition: 300ms all;
    color: ${colors.lila};
    ${TagLine} {
      transition: 300ms all;
      color: ${colors.lila};
      opacity: 1;
    }
  }
  @media (max-width: ${breakPoints.phone}) {
    display: block;
    margin-bottom: 16px;
  }
`;

export const Separator = styled.div`
  width: 24px;
  height: 1px;
  background-color: ${colors.grey};
  margin: 0 2rem;
  @media (max-width: ${breakPoints.phone}) {
    display: none;
  }
`;
