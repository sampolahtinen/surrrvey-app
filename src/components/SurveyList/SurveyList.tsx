import React from "react";
import {
  SurveyListContainer,
  Title,
  TagLine,
  Survey,
  Separator
} from "./SurveyList.styles";
import { SurveyListProps } from "./types";
import { useTrail } from "react-spring";
import { Link } from "react-router-dom";

const SurveyList: React.FC<SurveyListProps> = ({ surveys }) => {
  const config = { mass: 1, tension: 2000, friction: 200 };
  const trail = useTrail(surveys.length, {
    config,
    opacity: 1,
    from: { opacity: 0 }
  });
  return (
    <SurveyListContainer className="survey-list">
      {trail.map(({ opacity }, index) => (
        <Survey
          key={index}
          style={{ opacity }}
          to={`/surveys/${surveys[index].id}`}
        >
          <Title>{surveys[index].title}</Title>
          <Separator />
          <TagLine>{surveys[index].tagline}</TagLine>
        </Survey>
      ))}
    </SurveyListContainer>
  );
};

export default SurveyList;
