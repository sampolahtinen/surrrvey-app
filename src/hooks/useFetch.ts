import { useEffect, useState } from "react";

export function useFetch(
  callback: (params?: any) => Promise<any>,
  params?: any
) {
  const [results, setResults] = useState<any>({});
  const [error, setError] = useState(undefined);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const response = await callback(params);
        setResults(response.data);
        setIsLoading(false);
      } catch (error) {
        setIsLoading(false);
        setError(error);
      }
    };
    fetchData();
  }, [params, callback]);
  return [isLoading, results, error];
}
