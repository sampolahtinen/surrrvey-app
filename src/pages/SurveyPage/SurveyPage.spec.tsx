import React from "react";
import { shallow } from "enzyme";
import SurveysPage from "./SurveyPage";
import SurveyList from "../../components/SurveyList";

const mockSurveys = {
  surveys: [
    {
      id: "001",
      title: "test",
      tagline: "testtag"
    }
  ]
};

jest.mock("../../hooks/useFetch");
import { useFetch } from "../../hooks/useFetch";

describe("SurveyPage Component", () => {
  useFetch.mockReturnValue([true, mockSurveys, undefined]);
  const wrapper = shallow(<SurveysPage />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays h1", () => {
    expect(wrapper.find("h1").text()).toBe("Surveys");
  });
});

describe("SurveyPage displays SurveyList once loading is false", () => {
  useFetch.mockReturnValue([false, mockSurveys, undefined]);
  const wrapper = shallow(<SurveysPage />);

  it("displays SurveyList", () => {
    expect(wrapper.find(SurveyList).exists()).toBeTruthy();
  });
});
