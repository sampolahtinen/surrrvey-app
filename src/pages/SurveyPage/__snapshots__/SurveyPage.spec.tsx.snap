// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyPage Component renders 1`] = `
<Layout>
  <div
    className="surveys-page"
  >
    <h1
      style={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      Surveys
    </h1>
    <span />
  </div>
</Layout>
`;
